### PR TITLE
[ISSUE #3402]"getExtension()" can return null.[Trace]

### LINKED ISSUE #3402: [Enhancement] "getExtension()" can return null.[Trace]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
located at:
eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java line 123,143

### Describe the solution you'd like

image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by harshithasudhakar**
Fixes #3402 

### Motivation

Using requireNonNull() to check incoming references, can control the point in time when the exception will be thrown.

### Modifications

Included "Object.requireNonNull()"

### Documentation

- Does this pull request introduce a new feature? (yes / no)
 No
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
 No

### CODE REVIEW

**Review by Alonexc — COMMENTED** [review]
LGTM

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (xwm1992) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:123]: this dependent on `java.util.Objects` please import, and change it to `Objects.requireNonNull`.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (xwm1992) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:143]: same as above

**Review by xwm1992 — CHANGES_REQUESTED** [review]

**Review by mxsm — COMMENTED** [review]
Hi @harshithasudhakar  CI not pass, Please check it

**Comment by harshithasudhakar** [comment]
> Hi @harshithasudhakar CI not pass, Please check it

I have rectified the issue now.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (Alonexc) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:124]: is Objects.requireNonNull, not Object.requireNonNull.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (Alonexc) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:144]: is Objects.requireNonNull, not Object.requireNonNull.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (Alonexc) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:27]: This import is not in the right order and will result in checkstyle not passing:
image

**Review by Alonexc — DISMISSED** [review]
LGTM

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (pandaapo) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:124]: The judgment of ternary operation ensures that 'cloudEvent. getExtension (entry)' cannot return null here. `Objects.requireNonNull()` is redundant.

**Review by pandaapo — CHANGES_REQUESTED** [review]

**Review by pandaapo — CHANGES_REQUESTED** [review]
The judgment of ternary operation ensures that `cloudEvent. getExtension (entry)` cannot return null. `Objects.requireNonNull()` is redundant.

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (pandaapo) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:145]: It is OK now. The other one has the same problem. 
But in fact, this issue should not be posted as there is no problem in the original code here. So let's wait the maintainer to make the final decision.

Comment (pandaapo) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:145]: Now you see there is no problem with the original code here. The other one is in the same situation. 
In conclusion, this issue should not be posted. So let's wait the maintainer to make the final decision.

**Comment by scwlkq** [comment]
@pandaapo 你好👋 我想找一些初级的issue但是看这个貌似已经解决了但是没有关闭，这里还有其他直接commit的点吗？

hello 👋 I am looking for some beginner issues, but it seems that they have been resolved but not closed. Are there any other direct commit points here?

**Comment by mxsm** [comment]
> @pandaapo 你好👋 我想找一些初级的issue但是看这个貌似已经解决了但是没有关闭，这里还有其他直接commit的点吗？
> 
> hello 👋 I am looking for some beginner issues, but it seems that they have been resolved but not closed. Are there any other direct commit points here?

@scwlkq Hi you can try this issue #4502

**Review by mxsm — APPROVED** [review]

**Comment by Pil0tXia** [comment]
> What is the significance of these modifications in this PR? @mxsm

@pandaapo In your screenshots, it seems little changes have been made indeed.

**Comment by Pil0tXia** [comment]
Any progress can be made in this PR?

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (Pil0tXia) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:125]: Why do we need to inline the `addTraceInfoToSpan` method here? I think it is enough to add a `requireNonNull` method in the "add trace info" step of `addTraceInfoToSpan` method.

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java
Comment (pandaapo) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:125]: > I think it is enough to add a `requireNonNull` method in the "add trace info" step of `addTraceInfoToSpan` method.

I found that your current suggestion contradicts your previous one (https://github.com/apache/eventmesh/pull/3632#issuecomment-1776816518). So, why you now think it's needed to add an extra `requireNonNull` to `addTraceInfoToSpan` method?

Comment (Pil0tXia) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/trace/Trace.java:125]: Excuse me for overlooking some folded context earlier. I initially intended to point out that the `addTraceInfoToSpan` method should not be inlined here. However, it appears that further review is unnecessary as the issue corresponding to this PR is invalid. I will close the issue and label this PR as `invalid`.

@harshithasudhakar Thank you very much for your contribution. Please feel free to select other valuable issues to work on.

**Pull request closed without merging**
